### PR TITLE
[WIP] feature/team-builder

### PR DESCRIPTION
**Pull request by rootwizzy**

### CODE REVIEW

File: src/thinkspace/api/thinkspace-team/app/concerns/thinkspace/team/abstracts/team_set.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      pluck_to_hash(values, abstract_keys)
+    end
+
+    def get_assigned
+      # TODO: Switch to proper model and determine column based on presence of `transform` or not.
+      column = 'metadata'
+      key    = 'teams'
+      table  = 'thinkspace_readiness_assurance_responses'
+
+      query = "SELECT q3.id, q3.team_id, thinkspace_common_users.first_name, thinkspace_common_users.last_name FROM (SELECT q2.team_id AS team_id, (q2.users->>'id')::INTEGER AS id FROM (SELECT (q1.json1->>'id')::INTEGER AS team_id, jsonb_array_elements(q1.json1->'users') AS users FROM (SELECT json1 FROM #{table} t, jsonb_array_elements((t.#{column}->'#{key}')::jsonb) json1) q1) q2) q3 INNER JOIN thinkspace_common_users ON thinkspace_common_users.id = q3.id;"
```
Comment (rootwizzy) [inline src/thinkspace/api/thinkspace-team/app/concerns/thinkspace/team/abstracts/team_set.rb]: Breaking this large query string into a set of appended variables may improve maintainability if at some point we have to come back and edit this query. The example is just an idea of how we could set this up.

ex:
```
query  += "SELECT q3.id, q3.team_id, thinkspace_common_users.first_name, thinkspace_common_users.last_name "
query  += "FROM (SELECT q2.team_id AS team_id, (q2.users->>'id')::INTEGER AS id"
query  += "FROM (SELECT (q1.json1->>'id')::INTEGER AS team_id, jsonb_array_elements(q1.json1->'users') AS users "
query  += "FROM (SELECT json1 FROM #{table} t, jsonb_array_elements((t.#{column}->'#{key}')::jsonb) json1) q1) q2) q3 "
query  += "INNER JOIN thinkspace_common_users ON thinkspace_common_users.id = q3.id;"

```

File: src/thinkspace/api/thinkspace-team/app/models/thinkspace/team/team_set.rb [hunk 2]
```diff
@@ -89,6 +89,11 @@ def unlocked_teams
         thinkspace_team_teams.where.not(state: Thinkspace::Team::Team.state_locked)
       end
 
+      # ###
+      # ### Abstract
+      # ###
```
Comment (rootwizzy) [inline src/thinkspace/api/thinkspace-team/app/models/thinkspace/team/team_set.rb]: This is pretty close to what our new comment headers should look like with some minor changes.

`# ###` will convert any of the following text to a H3 which is fine if thats what you want the `Abstract` to be, but the wrapping `# ###` with no text will just appear as redudant hash marks in the source portion.

Just having the `# ### Abstract` will suffice but if you wish to append and prepend extra room just use a single `#`

```
#
# ### Abstract
#
```

or ideally just use the single line

**Review by rootwizzy — COMMENTED** [review]
Some minor requests for maintainability sakes. One thing I did not comment on is adding in the header and comment stubs. I saw there were a number of useful single line comments and a spattering of headers, but we should start to standardize this for all our files.

See [Comments Guide](http://totem-docs.herokuapp.com/1.0.0/code/comments) and [Documentation Types](http://totem-docs.herokuapp.com/1.0.0/code/documentation-types) for the proposed structure and [Lodestar Guides Generator](http://totem-docs.herokuapp.com/api/totem-lodestar/lib/tasks/totem/lodestar/helpers/api_generator.html) for a live example.

File: src/thinkspace/api/thinkspace-team/app/models/thinkspace/team/team_set.rb (hunk 2 above)
Comment (jamesfid) [reply src/thinkspace/api/thinkspace-team/app/models/thinkspace/team/team_set.rb]: Changed this.

File: src/thinkspace/api/thinkspace-team/app/concerns/thinkspace/team/abstracts/team_set.rb (hunk 1 above)
Comment (jamesfid) [reply src/thinkspace/api/thinkspace-team/app/concerns/thinkspace/team/abstracts/team_set.rb]: A potentially even better way would be using `%{}` for a multi-line string.  I will change it to that.

**Pull request merged**